Expensify/App
# Simplify TermsWithLicenses and SignInPageLayout

### Details
While fiddling around testing for https://github.com/Expensify/App/issues/4958, I discovered a large amount of duplicate code an unnecessarily divided implementations. This PR does some minor refactoring to simplify some redundant code in the signin/password pages. There's still more that could be done to clean this up, but this is a start.

### Fixed Issues
None.

### Tests / QA Steps
We just need to do regression testing of the signup/login pages – taking particular note of the Terms and Licenses at the bottom of the screen.

1. Go to the signup page.
1. Verify that the layout looks correct.
1. Verify that the terms and licenses at the bottom of the page are correctly formatted.
1. Go to the password page, and repeat the previous two steps.

### Tested On


### Screenshots
#### Web
![image](https://user-images.githubusercontent.com/47436092/131589824-b5d14777-8fdf-4721-979b-30af55f7014f.png)

#### Mobile Web
<img src="https://user-images.githubusercontent.com/47436092/131589985-0a705729-6c02-47a3-a540-1aae223ac185.png" alt="" width="400px" />

#### Desktop
<img src="https://user-images.githubusercontent.com/47436092/131590222-6ab4f334-8b8c-4381-8e85-3a8bab559967.png" alt="" width="400px" />

#### iOS
<img src="https://user-images.githubusercontent.com/47436092/131588799-c0837ac5-3a3c-4f07-ae10-f1675b7d900e.png" alt="" width="400px" />

#### Android
<img src="https://user-images.githubusercontent.com/47436092/131590569-d34c030c-a190-4867-b835-ac8e96d9efb8.png" alt="" width="400px" />
